### PR TITLE
Fix: Vela CLI 纳管集群后，无法通过APi接口对该集群创建namespace #4421

### LINKED ISSUE #4421: Vela CLI 纳管集群后，无法通过APi接口对该集群创建namespace
版本信息：
KubeVela 1.4.6
问题现象：
1、通过后台 CLI  vela cluster join 128config --name work001
image
2、通过APiserver 向该work001集群创建namespace时报错“cluster not found in data store”
image
3、根据解决方式 ，通过 list cluster 接口 同步数据，发现work001没有状态信息 
image
4、继续创建namespace,依然不行
image
5、登录Velaux界面，集群列表 ，点击编辑，后点击取消
image
image
集群显示健康
image
6、可以给集群创建namespace
image

### PR DESCRIPTION
**Pull request by sheldon-yan**
### Description of your changes

Fixes #4421

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by Somefive** [comment]
Is it a PR continuing #4505 ?

**Comment by sheldon-yan** [comment]
> Is it a PR continuing #4505 ?

Yes, I wrote the wrong PR title. I will revise it and resubmit it

**Comment by Somefive** [comment]
Looks good after nit. Please run `make reviewable`. Thanks!

File: pkg/apiserver/domain/service/cluster.go [hunk 1]
```diff
@@ -403,12 +403,22 @@ func (c *clusterServiceImpl) DeleteKubeCluster(ctx context.Context, clusterName
 }
 
 func (c *clusterServiceImpl) CreateClusterNamespace(ctx context.Context, clusterName string, req apis.CreateClusterNamespaceRequest) (*apis.CreateClusterNamespaceResponse, error) {
-	_, err := c.getClusterFromDataStore(ctx, clusterName)
+	cluster, err := c.getClusterFromDataStore(ctx, clusterName)
```
Comment (Somefive) [inline pkg/apiserver/domain/service/cluster.go:406]: let it be `_, err :=` here to fit the lint check

**Review by wonderflow — APPROVED** [review]
good job

**Comment by wonderflow** [comment]
it would be better if you could add some test cases

**Review by Somefive — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**